components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java

```diff
@@ -373,6 +382,68 @@ public RuleTemplate getRuleTemplate(String templateGroupUUID, String ruleTemplat
         return derivedTemplates;
     }
 
+    public Map<String, Artifact> deriveTemplates(BusinessRuleFromScratch businessRuleFromScratch) throws
+            TemplateManagerException {
```

Comment:
Can we check if this method also throws TemplateManagerException?

Reply:
This was fixed by df3bc5a752f0dfcd03393c5f62409814722e720b